### PR TITLE
feat/goldfish bridge

### PR DESCRIPTION
**Pull request by nikogu**
### TODO
- [x] Promise
- [ ] Mock
- [ ] Compatibility

### CODE REVIEW

File: packages/bridge/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// my.xxx type bridge
+export type BridgeType = Pick<typeof my, 'getAuthCode' | 'request'>;
```
Comment (nikogu) [inline packages/bridge/src/index.ts]: TODO：need to add picked

File: packages/bridge/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    return new Promise((resolve, reject) => {
+      if (typeof (my as any)[api] === 'function' && params) {
+        (my as any)[api]({
+          ...params,
+          success(res: PickSuccessResult<Parameters<R[T]>[0]>) {
+            resolve(res);
+          },
+          fail() {
+            reject();
+          },
```
Comment (yibuyisheng) [inline packages/bridge/src/index.ts]:
```ts
{
  fail: reject,
}
```

File: packages/bridge/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        });
+      }
+    });
+  },
+
+  // my.call('xxx')
+  mycall: async function mycall<T, R = Record<string, any>>(
```
Comment (yibuyisheng) [inline packages/bridge/src/index.ts]: T => R (`R`eturn)
R => P (`P`arams)

File: packages/bridge/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        params,
+        (result: T) => {
+          resolve(result);
+        },
+      );
+    });
+  },
+
+  // my.ap.xxx
+  ap: async function call<T extends keyof R, R extends Record<string, any> = APBridgeType>(
```
Comment (yibuyisheng) [inline packages/bridge/src/index.ts]: R any 同上。

**Pull request merged**